File: src/dto/messaging/message.DTO.ts
export type MessageDTO = {
  id: number;
  chatRoomId: number;
  senderId: number;
  senderName: string;
  senderAvatar?: string;
  content: string;
  type: 'TEXT' | 'IMAGE' | 'EMOJI' | 'LOCATION' | 'SYSTEM';
  imageUrl?: string;
  latitude?: number;
  longitude?: number;
  address?: string;
  createdAt: string;
  isRead: boolean;
};

export interface SendLocationMessageRequest extends SendMessageRequest {
  type: 'LOCATION';
  latitude: number;
  longitude: number;
  placeName: string;
  placeId?: string;
}

export type SendMessageRequest = {
  chatRoomId: number;
  content: string;
  type: 'TEXT' | 'IMAGE' | 'EMOJI' | 'LOCATION' | 'SYSTEM';
  imageUrl?: string;
  latitude?: number;
  longitude?: number;
  placeName?: string;
  placeId?: string;
};

export type ChatDTO = {
  id: number;
  type: 'PRIVATE' | 'GROUP';
  name?: string;
  participants: ChatParticipantDTO[];
  lastMessage?: MessageDTO;
  unreadCount: number;
  eventId?: number;
  pinnedMessage?: MessageDTO;
};

export type ChatParticipantDTO = {
  id: number;
  fullName: string;
  avatarUrl?: string;
  role: 'member' | 'organizer';
};


export type CreateChatRequest = {
  type: 'one_to_one' | 'group';
  participantIds: number[];
  name?: string;
  eventId?: number;
};

File: src/main/java/com/example/mobileApp/dto/messaging/MessageDTO.java
package com.example.mobileApp.dto.messaging;

import com.example.mobileApp.entity.ChatMessage.MessageStatus;
import com.example.mobileApp.entity.ChatMessage.MessageType;
import lombok.Data;
import java.time.LocalDateTime;

@Data
public class MessageDTO {
    private Long id;
    private Long senderId;
    private String senderName;
    private String senderAvatar;
    private Long chatRoomId;
    private String content;
    private MessageType type;
    private MessageStatus status;
    private LocalDateTime createdAt;

    // Location fields
    private Double latitude;
    private Double longitude;
    private String placeName;
    private Long locationId;
}
